unsloth/models/gemma.py

```diff
@@ -122,9 +127,12 @@ def GemmaDecoderLayer_fast_forward(
         hidden_states += residual
     else:
         residual = hidden_states
-        hidden_states = fast_rms_layernorm(
-            self.input_layernorm, hidden_states, gemma = True
-        )
+        if _DISABLE_TRITON_RMSNORM or _LAYERNORM_IMPL == "python":
+            hidden_states = self.input_layernorm(hidden_states)
+        else:
+            hidden_states = fast_rms_layernorm(
```

Comment:
Wait are you certain this causes NaN issues? That's extremely weird

Reply:
casting eps to float32 resolved that

<img width="774" height="570" alt="image" src="https://github.com/user-attachments/assets/b4e5df89-939b-407c-ab3e-d8c84b3ab2b6" />

but im still getting nans for 

```
m,tok = FastModel.from_pretrained(
"unsloth/gemma-3-4b-it",
load_in_4bit=False, load_in_8bit=False, full_finetuning=False,
)
m.train().cuda()
b = tok(text=["hello world"]*2, return_tensors="pt", padding=True).to("cuda")
out = m(**b, labels=b["input_ids"])
```